### DISCUSSION TITLE
Iterate through array request

### DISCUSSION THREAD

**Discussion by KalaiJay** [mockoon/mockoon#952] [Help] [answered]
Hi, 
I  need some one to guide me on how to iterate through an array request and get the response out of it. I tried using repeat and bodyRaw. I was not able to get the expected response. Below are my sample request and response. 

//Request
```
{
	"fruits": [{
		"name": "apple",
		"quantity": 30.000,
		"stores": [{
			"storeId": "01",
			"storename": "walmart"
		},
{			"storeId": "02",
			"storename": "kmart"
		}]
	}]
}
```

//Response

```
{ 
    "fruits": [
        {
            "name": "apple",
            "quantity": 30,
            "stores": [
                {
                    "storeId": "01",
                    "storename": "walmart",
                    "fruitprice": 120
                },
                {
                    "storeId": "02",
                    "storename": "kmart",
                    "fruitprice": 140
                }
            ]
        }
    ]
}
```

Below is my sample code in Mockoon which I tried. Can some one please help me on this?
```
{
    "fruits": [
        {
            {{#repeat 2}}
            "name": "{{body 'fruits.{{@index}}.name'}},",
            "quantity": {{body 'fruits.{{@index}}.quantity'}},
            "stores": [
                {
                    "storeId": "{{body 'fruits.' {{@index}} '.stores.0.storeId'}}",
                    "storename": "{{body 'fruits.{{@index}}.stores.0.storename'}}",
                    "fruitprice": {{add @index 10}}
                }
            ]
             {{/repeat}}
        }
    ]
}
```

**Comment by 255kb** [comment, marked as answer]
The Handlebars syntax is a bit cumbersome sometimes. First thing, you cannot imbricate curly braces in curly braces. Also, you cannot use `@index` inside a string, you will have to use [`concat`](https://mockoon.com/docs/latest/templating/mockoon-helpers/#concat). You can also use the `this` keyword inside an `each`. 
I am not sure what you are trying to achieve with the `repeat`, but I guess what you are looking for is the Handlebars' `each` helper which allows you to iterate on an array. 
Here is a working version, but I am not sure this is exactly waht you need. I hope it will help you solve your issue:

```
{
    "fruits": [
        {{#each (bodyRaw 'fruits')}}
        {
            "name": "{{body (concat 'fruits.' @index '.name')}}",
            "quantity": {{body (concat 'fruits.' @index '.quantity')}},
            "stores": [
                {{#each (bodyRaw (concat 'fruits.' @index '.stores'))}}
                {
                    "storeId": "{{this.storeId}}",
                    "storename": "{{this.storename}}",
                    "fruitprice": {{add @index 10}}
                }
                {{#unless @last}},{{/unless}}
                {{/each}}
            ]
        }
       {{/each}}
    ]
}
```

If you are going to create complex templates like this one, I suggest that you get familiar with the [Handlebars syntax](https://handlebarsjs.com/).

**Reply by KalaiJay** [reply]
Thanks a lot!. This is the one I was looking for.
